codebuild/create_project.py

```diff
@@ -319,20 +334,50 @@ def main(**kwargs):
         except exceptions.NoCredentialsError:
             raise SystemExit(f"Something went wrong with your AWS credentials;  Exiting.")
 
-        if args.dry_run:
-            logging.info('Respecting dry-run flag.  Done')
+        if args.production:
+            if args.modify_existing is True:
+                stack_name = config.get("Global", "stack_name")
+                change_set_name = "A" + hashlib.sha256(bytes(time.asctime().encode('utf-8'))).hexdigest()
+                # ^^^ Why are ChangeSetNames required to start with an Alphabetic character?
+
+                client.create_change_set(
+                    StackName=stack_name,
+                    TemplateBody=codebuild.to_yaml(),
+                    Capabilities=["CAPABILITY_IAM"],
+                    ChangeSetName=change_set_name)
+
+                logging.info("Waiting for change set {}".format(change_set_name))
+                waiter = client.get_waiter('change_set_create_complete')
+                waiter.wait(StackName=stack_name, ChangeSetName=change_set_name, WaiterConfig={"Delay": 3, "MaxAttempt": 3})
```

Comment:
What is the user experience like for the Changeset wait cycle? Does this throw an exception if the change set is invalid?  does the user need to cntrl-C if they decide they don't want to proceed?

Reply:
If the change set is invalid, depending on what is wrong an exception will be thrown during validation or when creating the change set. I've added another waiter which will fail if the change set fails to update.

I've also updated the code a bit to display changes and to give the user the option to cancel.

Reply:
Thanks